xbmc/dialogs/GUIDialogYesNo.cpp

```diff
@@ -65,12 +74,16 @@ bool CGUIDialogYesNo::OnBack(int actionID)
 {
   m_bCanceled = true;
   m_bConfirmed = false;
+  m_bCustomClicked = false;
   return CGUIDialogBoxBase::OnBack(actionID);
 }
 
 void CGUIDialogYesNo::OnInitWindow()
 {
-  SET_CONTROL_HIDDEN(CONTROL_CUSTOM_BUTTON);
+  if (m_bCustomVisible)
```

Comment:
if we rely on `m_strChoices[2]` there is no need to hold a bool for the visibility.

```c++
if (m_strChoices[2].empty())
  SET_CONTROL_HIDDEN(CONTROL_CUSTOM_BUTTON);
else
  SET_CONTROL_VISIBLE(CONTROL_CUSTOM_BUTTON);
```